mainapp/views.py

```diff
@@ -114,3 +114,24 @@ def handler500(request, *args, **argv):
     response = render(request, 'mainapp/error_handler.html')
     response.status_code = 500
     return response
+
+
+def SaveList(request):
+    "View to render to_read page"
+    user_ratings = list(UserRating.objects.filter(user=request.user).order_by('-bookrating'))
+    rated_books = set(get_rated_bookids(user_ratings))
+    book = set(SaveForLater.objects.filter(user=request.user).values_list('bookid', flat=True))
+    book_id = list(book)
+    for i in range(len(book_id)):
+        if book_id[i] in rated_books:
+            later = SaveForLater.objects.filter(user=request.user, bookid=book_id[i])
+            later.delete()
+    if len(book_id) == 0:
+        messages.info(request, 'Please Add Some Books')
+        return redirect('index')
+    books = get_book_dict(book_id)
```

Comment:
After deleting books from saved_book which are already rated the corresponding `book_id` should also be removed from the `book_id` list.